src/utils/colorConversion.ts

```diff
@@ -0,0 +1,33 @@
+const rgbSelector = /rgb([\s\S]*?)[)]/gs;
+
+// Converts a single string
+export const rgbToHex = (rgb: string) => {
+  const [r, g, b] = rgb.replace(/[^\d,.]/g, '').split(',');
+
+  return '#' + ((1 << 24) + (+r << 16) + (+g << 8) + +b).toString(16).slice(1);
```

Comment:
Converts using binary shifting - https://ryanclark.me/rgb-to-hex-via-binary-shifting/